src/poke_env/environment/target.py

```diff
@@ -0,0 +1,57 @@
+"""This module defines the Target class, which represents the types of
+targets a move can have
+"""
+
+import logging
+from enum import Enum, unique, auto
+
+# This is an enum for all the targets you can have
+@unique
+class Target(Enum):
```

Comment:
we probably want a method that converts (self, pokemon, battle) to a valid showdown target

Reply:
Great; I'll combine this with [Issue #523](https://github.com/hsahovic/poke-env/issues/523) then. 

I'll implement: `def to_showdown_target(self, order: BattleOrder, battle: AbstractBattle) -> int:`

It assumes in that DoubleBattleOrder(first_order, second_order) will map to first_order maps to the pokemon at 0th index in battle.active_pokemon and second_order maps to the pokemon at the 1st index in battle.active_pokemon

Reply:
I encountered a problem -- we'll have circular imports if we try to import Pokemon and Double Battle into the Target class, since Target is needed in Player, which imports Battle, which imports Move, which imports Target.

To remedy, I'll write this function in the DoubleBattle class as:
`def get_showdown_target(self, move: Move, target_mon: Pokemon) -> Optional[int]:`

(DoubleBattle, since targets are not needed for SingleBattle)